src/main/java/com/minecolonies/core/colony/managers/TravellingManager.java

```diff
@@ -0,0 +1,213 @@
+    {
+        travelerDataMap.put(citizenId, new TravelerData(citizenId, target, travelTimeInTicks));
+        colony.markDirty();
+    }
+
+    @Override
+    public void finishTravellingFor(final int citizenId)
+    {
+        if (travelerDataMap.containsKey(citizenId))
+        {
+            travelerDataMap.remove(citizenId);
+            colony.markDirty();
+        }
+    }
+
+    @Override
+    public void recallAllTravellingCitizens()
+    {
+        final List<ICitizenData> travellersToRecall = this.travelerDataMap
+          .keySet()
+          .stream()
+          .map(this.colony.getCitizenManager()::getCivilian)
+          .toList();
+
+        this.travelerDataMap.clear();
+        colony.markDirty();
+
+        travellersToRecall.forEach(citizenData -> {
+            final BlockPos spawnHutPos;
+            if (citizenData.getWorkBuilding() != null) {
+                spawnHutPos = citizenData.getWorkBuilding().getPosition();
+            }
+            else
+            {
+                spawnHutPos = colony.getBuildingManager().getTownHall().getPosition();
+            }
+
+            Optional<AbstractEntityCitizen> optionalEntityCitizen = citizenData.getEntity();
+            if (optionalEntityCitizen.isEmpty())
+            {
+                Log.getLogger().warn(String.format("The traveller %d from colony #%d has returned very confused!", citizenData.getId(), colony.getID()));
+                citizenData.setNextRespawnPosition(EntityUtils.getSpawnPoint(colony.getWorld(), spawnHutPos));
+                citizenData.updateEntityIfNecessary();
+                optionalEntityCitizen = citizenData.getEntity();
+            }
+
+            optionalEntityCitizen.ifPresent(abstractEntityCitizen -> TeleportHelper.teleportCitizen(abstractEntityCitizen, colony.getWorld(), spawnHutPos));
+        });
+    }
+
+    public boolean onTick()
+    {
+        travelerDataMap.values().forEach(TravelerData::onTick);
+        return true;
+    }
+
+    @Override
+    public CompoundTag serializeNBT()
+    {
+        final CompoundTag data = new CompoundTag();
+        final ListTag travelerData = new ListTag();
+
+        travelerDataMap.values().stream().map(TravelerData::serializeNBT).forEach(travelerData::add);
+
+        data.put(NbtTagConstants.TRAVELER_DATA, travelerData);
+
+        return data;
+    }
+
+    @Override
+    public void deserializeNBT(final CompoundTag nbt)
+    {
+        final ListTag travelerData = nbt.getList(NbtTagConstants.TRAVELER_DATA, Tag.TAG_COMPOUND);
+        travelerDataMap.clear();
+
+        travelerData
+          .stream()
+          .filter(CompoundTag.class::isInstance)
+          .map(CompoundTag.class::cast)
+          .map(TravelerData::new)
+          .forEach(data -> travelerDataMap.put(data.getCitizenId(), data));
+    }
+
+    private static final class TravelerData implements INBTSerializable<CompoundTag>
```

Comment:
I would almost say tha tthis should be its own file with doc

Reply:
still

Reply:
Why? this class is completely internally and not exposed anywhere in this classes API nor in the super interface type.

So I see no reason to do this.